index.js

```diff
@@ -47,26 +47,49 @@ EventResource.prototype.handle = function (ctx, next) {
         }
     }
   };
-
-  if (ctx.method === "POST" && this.events.post) {
-    this.events.post.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "GET" && this.events.get) {
-    this.events.get.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "DELETE" && this.events.delete) {
-    this.events.delete.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "PUT" && this.events.put) {
-    this.events.put.run(ctx, domain, function(err) {
-      ctx.done(err, result);
+  if (this.events.beforerequest !== undefined) {
```

Comment:
You should just check for `if (this.events.beforerequest) {`

see line 54, 58, 62, etc.